sklearn/metrics/tests/test_classification.py

```diff
@@ -348,6 +351,133 @@ def test_precision_recall_f_ignored_labels():
 
 
+def test_tpr_fpr_tnr_fnr_scores_binary_averaged():
+    # Test TPR, FPR, TNR, FNR scores for binary classification task
+    y_true, y_pred, _ = make_prediction(binary=True)
+
+    # compute scores with default labels introspection
+    tprs, fprs, tnrs, fnrs = tpr_fpr_tnr_fnr_scores(y_true, y_pred, average=None)
+    assert_array_almost_equal(tprs, [0.88, 0.68], 2)
+    assert_array_almost_equal(fprs, [0.32, 0.12], 2)
+    assert_array_almost_equal(tnrs, [0.68, 0.88], 2)
+    assert_array_almost_equal(fnrs, [0.12, 0.32], 2)
+
+    tn, fp, fn, tp = assert_no_warnings(confusion_matrix, y_true, y_pred).ravel()
+    assert_array_almost_equal(tp / (tp + fn), 0.68, 2)
+    assert_array_almost_equal(fp / (tn + fp), 0.12, 2)
+    assert_array_almost_equal(tn / (tn + fp), 0.88, 2)
+    assert_array_almost_equal(fn / (tp + fn), 0.32, 2)
+
+    tpr, fpr, tnr, fnr = tpr_fpr_tnr_fnr_scores(y_true, y_pred, average="macro")
+    assert tpr == np.mean(tprs)
+    assert fpr == np.mean(fprs)
+    assert tnr == np.mean(tnrs)
+    assert fnr == np.mean(fnrs)
+
+    tpr, fpr, tnr, fnr = tpr_fpr_tnr_fnr_scores(y_true, y_pred, average="weighted")
+    support = np.bincount(y_true)
+    assert tpr == np.average(tprs, weights=support)
+    assert fpr == np.average(fprs, weights=support)
+    assert tnr == np.average(tnrs, weights=support)
+    assert fnr == np.average(fnrs, weights=support)
+
+
+def test_tpr_fpr_tnr_fnr_scores_multiclass():
+    # Test TPR, FPR, TNR, FNR scores for multiclass classification task
+    y_true, y_pred, _ = make_prediction(binary=False)
+
+    # compute scores with default labels introspection
+    tprs, fprs, tnrs, fnrs = tpr_fpr_tnr_fnr_scores(y_true, y_pred, average=None)
+    assert_array_almost_equal(tprs, [0.79, 0.1, 0.9], 2)
+    assert_array_almost_equal(fprs, [0.08, 0.14, 0.45], 2)
+    assert_array_almost_equal(tnrs, [0.92, 0.86, 0.55], 2)
+    assert_array_almost_equal(fnrs, [0.21, 0.9, 0.1], 2)
+
+    # averaging tests
+    tpr, fpr, tnr, fnr = tpr_fpr_tnr_fnr_scores(y_true, y_pred, average="micro")
+    assert_array_almost_equal(tpr, 0.53, 2)
+    assert_array_almost_equal(fpr, 0.23, 2)
+    assert_array_almost_equal(tnr, 0.77, 2)
+    assert_array_almost_equal(fnr, 0.47, 2)
+
+    tpr, fpr, tnr, fnr = tpr_fpr_tnr_fnr_scores(y_true, y_pred, average="macro")
+    assert_array_almost_equal(tpr, 0.6, 2)
+    assert_array_almost_equal(fpr, 0.22, 2)
+    assert_array_almost_equal(tnr, 0.78, 2)
+    assert_array_almost_equal(fnr, 0.4, 2)
+
+    tpr, fpr, tnr, fnr = tpr_fpr_tnr_fnr_scores(y_true, y_pred, average="weighted")
+    assert_array_almost_equal(tpr, 0.53, 2)
+    assert_array_almost_equal(fpr, 0.2, 2)
+    assert_array_almost_equal(tnr, 0.8, 2)
+    assert_array_almost_equal(fnr, 0.47, 2)
+
+    with pytest.raises(ValueError):
+        tpr_fpr_tnr_fnr_scores(y_true, y_pred, average="samples")
+
+    # same prediction but with and explicit label ordering
```

Comment:
The coverage is at 100%.

Reply:
> It makes me think that we should check the coverage. I don't see the report in the last push. It could be nice to be sure that we covered all the branch of the code.

The coverage is proudly at 100%.

Reply:
> Could we add a case where we take a subset of the labels.

Done.

Reply:
> We should probably as well test when there is too many labels given

Done.